### PR TITLE
grafanaPlugins.yesoreyeram-infinity-datasource: init at 2.8.0

### PR DESCRIPTION
**Pull request by newAM**
## Description of changes

Add the infinity data source: <https://grafana.com/grafana/plugins/yesoreyeram-infinity-datasource>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ grafanaPlugin, lib }:
+
+grafanaPlugin {
+  pname = "yesoreyeram-infinity-datasource";
+  version = "2.6.3";
+  zipHash = {
```
Comment (Pamplemousse) [inline pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix:6]: I am curious: how did you get the `zipHash`es for the platforms that you aren't using?

File: pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    x86_64-linux = "sha256-Se0leqiLnPdHbaJkwN70mY5GK5lAllZCUrFc8oLgrSc=";
+    aarch64-linux = "sha256-r9ObUaN8lF6S5p+ZjhOQ0KcSNgrbzya3PbKr+MlCly0=";
+    x86_64-darwin = "sha256-TpyKqIGBxYM6edavYKbqBd2Kdz3UkST7kP90e69XY2s=";
+    aarch64-darwin = "sha256-2QK5nypMdJ8lONGsrKqqR0oNnmcHGTD7mTazFv7AxCM=";
+  };
+  meta = with lib; {
+    description = "CSV, JSON, GraphQL, XML and HTML datasource for grafana.";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ newam ];
```
Comment (Pamplemousse) [inline pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix:15]: You can also add me as a maintainer.
I currently use https://git.sr.ht/~pamplemousse/nix-utils/tree/main/item/pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix, but would be happier if it was in `nixpkgs` :+1:

**Comment by Pamplemousse** [comment]
LGTM :+1:

**Review by Pamplemousse — APPROVED** [review]

File: pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix (hunk 1 above)
Comment (newAM) [reply pkgs/servers/monitoring/grafana/plugins/yesoreyeram-infinity-datasource/default.nix:6]: I downloaded the zip files with `wget`, ran a `sha256sum` on it, then a `nix hash to-sri --type sha256` to get it into SRI format.

**Review by Pamplemousse — APPROVED** [review]

**Comment by newAM** [comment]
CC grafana maintainers, @fpletz @willibutz @globin @ma27 @Frostman, I would appreciate a review if any of you have the time! :)

**Review by Pamplemousse — APPROVED** [review]

**Review by Pamplemousse — APPROVED** [review]

**Comment by newAM** [comment]
The package has moved on to 2.9.5.  I have updated this PR twice with the latest version, but it doesn't seem there is interest in merging.  Going to close.

**Comment by Pamplemousse** [comment]
https://github.com/NixOS/nixpkgs/pull/264566#issuecomment-2030298589

I am still "maintaining" the plugins I use over there: https://git.sr.ht/~pamplemousse/nix-utils/tree/main/item/pkgs/servers/monitoring/grafana/plugins . Some might be quite outdated, but I accept patches :+1:

**Pull request closed without merging**